src/Sharpliner.Model/Job.cs

```diff
@@ -3,20 +3,40 @@
 
 namespace Sharpliner.Model
 {
-    public record Job(
-        string Name,
-        string DisplayName,
-        IEnumerable<string> DependsOn,
-        string Condition,
-        JobStrategy Strategy,
-        JobPool Pool,
-        ContainerReference Container,
-        TimeSpan Timeout,
-        TimeSpan CancelTimeout,
-        IEnumerable<Variable> Variables,
-        IEnumerable<Step> Steps,
-        JobWorkspace Workspace = JobWorkspace.Outputs,
-        bool ContinueOnError = false);
 
     // TODO: missing properties Uses, Services
+    public record Job
+    {
+        public string Name { get; }
+
+        public string DisplayName { get; }
+
+        public IEnumerable<string>? DependsOn { get; init; }
+
+        public string? Condition { get; init; }
+
+        public JobStrategy? Strategy { get; init; }
+
+        public Pool? Pool { get; init; }
+
+        public ContainerReference? Container { get; init; }
+
+        public TimeSpan? Timeout { get; init; }
+
+        public TimeSpan? CancelTimeout { get; init; }
+
+        public IEnumerable<Variable>? Variables { get; init; }
+
+        public IEnumerable<Step>? Steps { get; init; }
```

Comment:
This is going to have to be a list and be initialize already. That way users just have to use {} which results in a call to .Add
```suggestion
        public List<Step> Steps { get; } = new List<Step>();
```
If you try the API as you wrote it the user will have to do:
```csharp 
Steps = new [] {}
// or
Steps = new List<Step> ();
// But if the list is already present
Steps = {
new Step()
}
```

Reply:
Oh my god! This is amazing! I didn't know about this! I was always wondering how we were gonna achieve `{}` only